neuro_san/service/grpc/grpc_agent_server.py

```diff
@@ -104,35 +105,37 @@ def get_services(self) -> List[GrpcAgentService]:
         """
         return self.services
 
-    def agent_added(self, agent_name: str):
+    def agent_added(self, agent_name: str, source: AgentStorageSource):
         """
         Agent is being added to the service.
         :param agent_name: name of an agent
+        :param source: The AgentStorageSource source of the message
         """
-        public_storage: AgentNetworkStorage = self.network_storage_dict.get("public")
         service = GrpcAgentService(self.server_lifetime, self.security_cfg,
                                    agent_name,
-                                   public_storage.get_agent_network_provider(agent_name),
+                                   source.get_agent_network_provider(agent_name),
```

Comment:
Use the source to get the AgentNetworkProvider.

Reply:
Yes.  With only one source possible right now, we don't need such verification, but I certainly get where you are going here.  Can address in future PRs if necessary.